x/emissions/migrations/v3/migrate_test.go

```diff
@@ -0,0 +1,72 @@
+package v3_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	keepertest "github.com/zeta-chain/zetacore/testutil/keeper"
+	"github.com/zeta-chain/zetacore/x/emissions/exported"
+	v3 "github.com/zeta-chain/zetacore/x/emissions/migrations/v3"
+	"github.com/zeta-chain/zetacore/x/emissions/types"
+)
+
+type mockSubspace struct {
+	ps types.Params
+}
+
+func newMockSubspace(ps types.Params) mockSubspace {
+	return mockSubspace{ps: ps}
+}
+
+func (ms mockSubspace) GetParamSet(ctx sdk.Context, ps exported.ParamSet) {
+	*ps.(*types.Params) = ms.ps
+}
+
+func TestMigrate(t *testing.T) {
+	t.Run("should migrate for valid params", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.EmissionsKeeper(t)
+
+		legacyParams := types.Params{
+			MaxBondFactor:               "1",
+			MinBondFactor:               "0.75",
+			AvgBlockTime:                "5.00",
+			TargetBondRatio:             "00.50",
+			ValidatorEmissionPercentage: "00.50",
+			ObserverEmissionPercentage:  "00.35",
+			TssSignerEmissionPercentage: "00.15",
+			DurationFactorConstant:      "0.001877876953694702",
+			ObserverSlashAmount:         sdk.ZeroInt(),
+		}
+		legacySubspace := newMockSubspace(legacyParams)
+
+		require.NoError(t, v3.MigrateStore(ctx, k, legacySubspace))
+
+		params, found := k.GetParams(ctx)
+		require.True(t, found)
+		legacyParams.ObserverSlashAmount = sdkmath.NewInt(100000000000000000)
+		legacyParams.BallotMaturityBlocks = 100
+		require.Equal(t, legacyParams, params)
+	})
+
+	t.Run("should fail to migrate for invalid params", func(t *testing.T) {
+		k, ctx, _, _ := keepertest.EmissionsKeeper(t)
+
+		legacyParams := types.Params{
+			MaxBondFactor:               "1",
+			MinBondFactor:               "0.50",
+			AvgBlockTime:                "5.00",
+			TargetBondRatio:             "00.50",
+			ValidatorEmissionPercentage: "00.50",
+			ObserverEmissionPercentage:  "00.35",
+			TssSignerEmissionPercentage: "00.15",
+			DurationFactorConstant:      "0.001877876953694702",
+			ObserverSlashAmount:         sdk.ZeroInt(),
+		}
+		legacySubspace := newMockSubspace(legacyParams)
+
+		require.Error(t, v3.MigrateStore(ctx, k, legacySubspace))
```

Comment:
Maybe we can use `ErrorIs`  and check for the error from the validate function ?